exporter/cassandraexporter/exporter_logs.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandraexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/cassandraexporter"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gocql/gocql"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
+)
+
+type logsExporter struct {
+	client *gocql.Session
+	logger *zap.Logger
+	cfg    *Config
+}
+
+func newLogsExporter(logger *zap.Logger, cfg *Config) (*logsExporter, error) {
+	cluster := gocql.NewCluster(cfg.DSN)
+	session, err := cluster.CreateSession()
+	cluster.Keyspace = cfg.Keyspace
+	cluster.Consistency = gocql.Quorum
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &logsExporter{logger: logger, client: session, cfg: cfg}, nil
+}
+
+func initializeLogKernel(cfg *Config) error {
+	ctx := context.Background()
+	cluster := gocql.NewCluster(cfg.DSN)
+	cluster.Consistency = gocql.Quorum
+	session, err := cluster.CreateSession()
+	if err != nil {
+		return err
+	}
+
+	createDatabaseError := session.Query(parseCreateDatabaseSQL(cfg)).WithContext(ctx).Exec()
+	if createDatabaseError != nil {
+		return createDatabaseError
+	}
+	createLogTableError := session.Query(parseCreateLogTableSQL(cfg)).WithContext(ctx).Exec()
+	if createLogTableError != nil {
+		return createLogTableError
+	}
+
+	defer session.Close()
```

Comment:
This should go right after 
```
	session, err := cluster.CreateSession()
	if err != nil {
		return err
	}
```
Otherwise it won't be closed in case of `createDatabaseError` or `createLogTableError`